### ISSUE TITLE
missing CRDs cause crash on startup instead of failure on replication creation

### ISSUE THREAD

**Issue by oriyarde** [csi-addons/volume-replication-operator#97]
the list of sidecars of our CSI controller deployment is constant, i.e. doesn't change according to existence of CRDs on the cluster.

so we would like to be able to deploy the volume replication operator even when the [replication CRDs](https://github.com/csi-addons/volume-replication-operator/tree/v0.1.0/config/crd/bases) don't exist on the cluster.
this is the error we get on startup:
```
2021-08-04T12:22:20.018Z        ERROR   controller-runtime.source       if kind is a CRD, it should be installed before calling Start   {"kind": "VolumeReplication.replication.storage.openshift.io", "error": "no matches for kind \"VolumeReplication\" in version \"replication.storage.openshift.io/v1alpha1\""}
github.com/go-logr/zapr.(*zapLogger).Error
        /go/pkg/mod/github.com/go-logr/zapr@v0.2.0/zapr.go:132
sigs.k8s.io/controller-runtime/pkg/source.(*Kind).Start
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.7.0/pkg/source/source.go:117
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.7.0/pkg/internal/controller/controller.go:159
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.7.0/pkg/internal/controller/controller.go:205
sigs.k8s.io/controller-runtime/pkg/manager.(*controllerManager).startRunnable.func1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.7.0/pkg/manager/internal.go:691
2021-08-04T12:22:20.019Z        ERROR   setup   problem running manager {"error": "no matches for kind \"VolumeReplication\" in version \"replication.storage.openshift.io/v1alpha1\""}
github.com/go-logr/zapr.(*zapLogger).Error
        /go/pkg/mod/github.com/go-logr/zapr@v0.2.0/zapr.go:132
main.main
        /workspace/main.go:134
runtime.main
        /usr/local/go/src/runtime/proc.go:204
```
note that e.g. the snapshotter [doesn't crash](https://github.com/kubernetes-csi/external-snapshotter/issues/212) when [its CRDs](https://github.com/kubernetes-csi/external-snapshotter/tree/v4.1.1/client/config/crd) are missing, it only errors when trying to create a snapshot, as desired.

**Comment by Madhu-1** [comment]
@oriyarde this is how the operator built from operator SDK works https://github.com/operator-framework/operator-sdk/issues/183#issuecomment-439484798

**Comment by oriyarde** [comment]
@Madhu-1 I don't think other sidecars (e.g. said snapshotter) are built from operator SDK, but a workaround might be to wait for the CRDs before [starting the manager](https://github.com/csi-addons/volume-replication-operator/blob/00f42351bccb8020ef90d36e65d4320959154285/main.go#L154)

**Comment by Madhu-1** [comment]
We are not completely following the sidecar model for building/developing volume replication it's purely built from the operator SDK framework.

AFAIC waiting at the operator level is not of much use. This is a process of installation instead of running the operator first the admin can install the CRD and then start the operator. or whoever is automating the operator installation check for the CRD exists in a loop once he sees the CRD then install the volume-replication operator.

**Comment by oriyarde** [comment]
is there a reason not to follow the sidecar model?
since this project is used as a sidecar, I think it would be simpler to wait for the CRDs before starting rather than require all driver operators to dynamically change deployment according to CRDs installation status at any given moment

**Comment by Madhu-1** [comment]
fixed in #108
